conformance/binary_json_conformance_suite.cc

```diff
@@ -2926,6 +2926,56 @@ void BinaryAndJsonConformanceSuite::RunJsonTestsForFieldMask() {
                                 R"(optional_field_mask: {paths: "foo_3_bar"})");
   ExpectSerializeFailureForJson("FieldMaskTooManyUnderscore", RECOMMENDED,
                                 R"(optional_field_mask: {paths: "foo__bar"})");
+
+  // https://google.aip.dev/161#map-fields introduces map keys as part of FieldMask path
+  // elements. If map keys contain characters otherwise problematic for FieldMask,
+  // backticks can be used for quoting. Backtick itself is escaped using
+  // double-backtick.
+  // If a path element is quoted, no camelCase conversion will be done when serializing
+  // to JSON, it will be copied to JSON as-is, including the quotes.
+  RunValidJsonTest("FieldMaskWithBacktickQuotes", RECOMMENDED,
+                   R"json({"optionalFieldMask": "`foo`"})json",
+                   R"(optional_field_mask: {paths: "`foo`"})");
+
+  RunValidJsonTest("FieldMaskWithBacktickQuotesNoCamelCase", RECOMMENDED,
+                   R"json({"optionalFieldMask": "`foo_bar`"})json",
+                   R"(optional_field_mask: {paths: "`foo_bar`"})");
+
+  RunValidJsonTest("FieldMaskWithBacktickQuotesTwoPaths", RECOMMENDED,
+                   R"json({"optionalFieldMask": "`foo`,`bar`"})json",
```

Comment:
One potentially-interesting case which I don't think is directly covered yet:

```text
JSON: "`foo,bar`"
Result: { paths: "`foo,bar`" }
```

I *suspect* the backtick should only occur *either* at the start and end of a path *or* double in an escape sequence. Although I'm not sure about that right now...

So we could have a bunch of test cases for "this can't be converted to JSON" or "this JSON can't be parsed".